config/waagent.logrotate

```diff
@@ -1,7 +1,22 @@
 /var/log/waagent.log {
+    # Old versions of log files are compressed with gzip by default.
     compress
-    monthly
-    rotate 6
+
+    # Rotate log files > 20 MB, and keep last 50 archived files. With a compression ratio of around 10:1, The archived
+    # files would be take an average of around 100 MB
+    size 20M
```

Comment:
how often are those files checked? I've seen logs > 1 GB... 50 may be too many

Reply:
`logrotate` runs once a day. The `logrotate's` cron file is kept in `/etc/cron.daily/`.

The 50 here would be the compressed files, and not uncompressed.

Reply:
but still, how well do those huge files (> 1GB) compress? i think i have some samples from incidents i've looked at, let me see if i can find them

Reply:
Some tests results - 

Uncompressed | Compressed | Compression Ratio
-- | -- | --
3102345801 | 154144402 | 4.97%
393461760 | 19558242 | 4.97%
33047881 | 1770543 | 5.36%
588946964 | 27436410 | 4.66%

Reply:
Tested with more chatty agents as well. With the size limit and daily logrotate in place, the files would be rotated every day, once they cross the 20 MB threshold.

```
size (bytes)
   1538167      test-1day.log.gz         # Compressed 1-day log.
  31809193      waagent.log.1-1day.log   # Compressed 1-day log.  (eg: 06/01 06:25:05 - 06/02 06:25:19)
 952814059      waagent.log.1.log        # Uncompressed 1-month log.
```

Reply:
Can we increase the cadence of logrotate to 2-3times a day? If the file size < 20MB it wont do anything and if its >20Mb it'll just rotate it. 
Does the logrotate process use a lot of resources? Any idea?

Reply:
The `logrotate` is a common utility and in most cases, `logrotate` is invoked from a script in the `/etc/cron.daily/` directory, which we don't have control over.

And if in the case where it is <20 MB for a long time, it'll remain unrotated - which isnt a problem. We want to catch the cases where we go to 40/50 MB per day usage and compress them immediately. In the previous case, we could go to 950MB for a month, without any rotation.

Reply:
Oh damn I didnt know about the 950mb part, that's insane! 
Thanks for adding it!